gm4_block_compressors/data/gm4_block_compressors/functions/item.mcfunction

```diff
@@ -1,13 +1,13 @@
 # @s = item above block compressor
 # at location of dropper (compressor)
 # run from process
+# tries to compress/decompress the current item entity.
+# action's states are 0:compress, 1:nothing, 2:attempted decompress
 
-data modify storage gm4_block_compressors:temp/compressor Items set from block ~ ~ ~ Items
 data modify storage gm4_block_compressors:temp/item_stack Item set from entity @s Item
 
-execute if data storage gm4_block_compressors:temp/compressor {Items:[{Count:64b}]} if data storage gm4_block_compressors:temp/item_stack Item{Count:64b} unless data storage gm4_block_compressors:temp/item_stack Item.tag.gm4_compressed run function gm4_block_compressors:compress/64
-execute if data storage gm4_block_compressors:temp/compressor {Items:[{Count:16b}]} if data storage gm4_block_compressors:temp/item_stack Item{Count:16b} unless data storage gm4_block_compressors:temp/item_stack Item.tag.gm4_compressed run function gm4_block_compressors:compress/16
-execute if data storage gm4_block_compressors:temp/compressor {Items:[{Count:1b}]} if data storage gm4_block_compressors:temp/item_stack Item{Count:1b,tag:{gm4_compressed:{value:64}}} run function gm4_block_compressors:decompress/64
-execute if data storage gm4_block_compressors:temp/compressor {Items:[{Count:1b}]} if data storage gm4_block_compressors:temp/item_stack Item{Count:1b,tag:{gm4_compressed:{value:16}}} run function gm4_block_compressors:decompress/16
+execute if data storage gm4_block_compressors:temp/item_stack Item{Count:1b} run function gm4_block_compressors:item/try_decompress
+execute unless score action gm4_bc_data matches 2 unless data storage gm4_block_compressors:temp/item_stack Item.tag.gm4_block_compressors unless data storage gm4_block_compressors:temp/item_stack gm4_compressed run function gm4_block_compressors:item/try_compress
```

Comment:
The `action gm4_bc_data` score doesn't make sense semantically anymore. Consider changing it to a simple boolean `$compressed` and you can store the success of the command on line 9.